### PR TITLE
Bump server dependencies to pre-release

### PR DESCRIPTION
**Pull request by scottn12**
This PR bumps server dependencies to 0.1036.3000-0 pre-release. This PR also updates usages of `IDocumentStorage.createDocument()` to prevent build errors (see https://github.com/microsoft/FluidFramework/pull/10161#issuecomment-1118830869).

### CODE REVIEW

**Review by ssimic2 — APPROVED** [review]

**Comment by ssimic2** [comment]
@scottn12 before merging, please check that `IDocumentStorage.createDocument()` changes are valid.

**Comment by scottn12** [comment]
@tianzhu007, can you please check that the changes related to `IDocumentStorage.createDocument()` are valid?

**Comment by tianzhu007** [comment]
> @tianzhu007, can you please check that the changes related to `IDocumentStorage.createDocument()` are valid?

Looks good!

File: server/routerlicious/packages/local-server/src/packageVersion.ts [hunk 1]
```diff
@@ -6,4 +6,4 @@
  */
 
 export const pkgName = "@fluidframework/server-local-server";
-export const pkgVersion = "0.1036.2000";
+export const pkgVersion = "0.1036.3000";
```
Comment (tylerbutler) [inline server/routerlicious/packages/local-server/src/packageVersion.ts:9]: Is this correct? You're consuming a pre-release of this version in this change. Don't you have to publish the server prerelease before you can make the changes in the local-driver?

Comment (tylerbutler) [reply server/routerlicious/packages/local-server/src/packageVersion.ts:9]: Oh, is this because the server change was actually in .2000?

**Pull request merged**